staging/src/k8s.io/apiserver/pkg/storage/cacher/delegator.go

```diff
@@ -33,18 +40,46 @@ import (
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	"k8s.io/component-base/tracing"
 	"k8s.io/klog/v2"
+	"k8s.io/utils/clock"
+)
+
+var (
+	// ConsistencyCheckPeriod is the period of checking consistency between etcd and cache.
+	// 5 minutes were proposed to match the default compaction period. It's magnitute higher than
+	// List latency SLO (30 seconds) and timeout (1 minute).
+	ConsistencyCheckPeriod = 5 * time.Minute
+	// ConsistencyCheckDelay is the initial delay of checking consistency between etcd and cache.
+	// Delaying the check prevents overlap with other operations at the apiserver start.
+	ConsistencyCheckDelay = time.Minute
 )
 
 func NewCacheDelegator(cacher *Cacher, storage storage.Interface) *CacheDelegator {
-	return &CacheDelegator{
+	return newCacheDelegator(cacher, storage, clock.RealClock{})
+}
+
+func newCacheDelegator(cacher *Cacher, storage storage.Interface, clock clock.Clock) *CacheDelegator {
+	d := &CacheDelegator{
 		cacher:  cacher,
 		storage: storage,
+		checker: newConsistencyChecker(cacher.resourcePrefix, cacher.newListFunc, cacher, storage, clock),
```

Comment:
Actually - one more comment. This shouldn't be done unconditionally.
We need to be able to switch it on/off - I don't want this to be running in all my production clusters...

Reply:
I missed gating the feature under feature gate, however this feature was intended for a Beta default in 1.33. For now it just exposes metric and doesn't do any actuation, it's intended to soak in production before we decide on actuation mechanism. 

What makes you think you will not run it in production?

Reply:
> now it just exposes metric and doesn't do any actuation, 

No - it actually lists and compares the results.

I don't want periodic listing in production clusters. The goal is to run this only in CI for now, not in millions of k8s clusters across the world.
We should have a mechanism to explicitly enable it the same we do for other kinds of checkers (like the one for watch-list, mutation detector, etc.)

Reply:
> No - it actually lists and compares the results.

That's the detection part, actuation would disable serving results from snapshots. 
I think there might be misunderstanding in the KEP, my plan was to run it in all cluster without actuation enabled and later add actuation when we enable serving requests from snapshot.

cc @deads2k

Reply:
>  my plan was to run it in all cluster without actuation enabled 

Then I don't agree with that. I don't want to enable it for all production clusters we have for sure.

Reply:
For Alpha, let's do your suggestion.